launcher/ui/dialogs/skins/draw/SkinOpenGLWindow.cpp

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2024 Trial97 <alexandru.tripon97@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "ui/dialogs/skins/draw/SkinOpenGLWindow.h"
+
+#include <QMouseEvent>
+#include <QOpenGLBuffer>
+#include <QVector2D>
+#include <QVector3D>
+#include <QtMath>
+
+#include "minecraft/skins/SkinModel.h"
+#include "rainbow.h"
+#include "ui/dialogs/skins/draw/BoxGeometry.h"
+#include "ui/dialogs/skins/draw/Scene.h"
+
+SkinOpenGLWindow::SkinOpenGLWindow(SkinProvider* parent, QColor color)
+    : QOpenGLWindow(), QOpenGLFunctions(), m_baseColor(color), m_parent(parent)
+{
+    QSurfaceFormat format = QSurfaceFormat::defaultFormat();
+    format.setDepthBufferSize(24);
+    setFormat(format);
+}
+
+SkinOpenGLWindow::~SkinOpenGLWindow()
+{
+    // Make sure the context is current when deleting the texture
+    // and the buffers.
+    makeCurrent();
+    // double check if resources were initialized because they are not
+    // initialized together with the object
+    if (m_scene) {
+        delete m_scene;
+    }
+    if (m_background) {
+        delete m_background;
+    }
```

Comment:
Btw just a tip: you can do this and it's still safe as delete nullptr is a no-op
```suggestion
    delete m_scene;
    delete m_background;
```